### PR TITLE
feat: Adds the temporal fusion transformer (TFT) forecasting job

### PR DESCRIPTION
**Pull request by mikelawrence-google**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-aiplatform/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes b/239609625 🦕

### CODE REVIEW

**Comment by mikelawrence-google** [comment]
@sasha-gitg Hi there, I need a little help and I see you've done a lot of work in this repo. Do you often have this presubmit issue with pytest like,

```
_____________________________ ERROR collecting gw2 _____________________________
Different tests were collected between gw3 and gw2. The difference is:
```

I don't think my PR has done anything particularly wild... I'm not sure how to fix this. Any ideas? If you think someone else might be better to ask, feel free to let me know

**Comment by sasha-gitg** [comment]
@mikelawrence-google Looks like an issue with python-xdist and `@pytest.mark.parametrize` with a set: https://github.com/pytest-dev/pytest-xdist/issues/149

Try changing back to list or following the sorted recommendation in the issue.

**Comment by mikelawrence-google** [comment]
Thanks for taking a look, @sasha-gitg and @sararob. I thought it was more related to build infrastructure, but it was indeed the set vs list issue. My team is waiting on some documentation before we can merge, but will get the review going

File: google/cloud/aiplatform/schema.py
Comment (geraint0923) [inline google/cloud/aiplatform/schema.py:27]: @mikelawrence-google can we confirm this schema is published and can be accessed?

**Review by geraint0923 — COMMENTED** [review]
Thanks for adding the SDK support for TFT!

File: google/cloud/aiplatform/schema.py
Comment (mikelawrence-google) [reply google/cloud/aiplatform/schema.py:27]: Yep, looks good from my side.

```
$ gsutil cat gs://google-cloud-aiplatform/schema/trainingjob/definition/temporal_fusion_transformer_time_series_forecasting_1.0.0.yaml
type: TemporalFusionTransformerTimeSeriesForecasting
version: 1.0.0
description: >
  A training task that trains a Temporal Fusion Transformer (TFT) Forecasting
  Model via the
...
...
```

**Review by geraint0923 — APPROVED** [review]

**Review by TheMichaelHu — APPROVED** [review]

File: google/cloud/aiplatform/training_jobs.py
Comment (sasha-gitg) [inline google/cloud/aiplatform/training_jobs.py:5386]: Docstring. What's the difference between this and the superclass `run`? If there is none, I think this can be removed.

**Review by sasha-gitg — CHANGES_REQUESTED** [review]

File: google/cloud/aiplatform/training_jobs.py
Comment (mikelawrence-google) [reply google/cloud/aiplatform/training_jobs.py:5386]: Thanks Sasha. For some background, I based this PR on the [Seq2SeqPlus PR](https://github.com/googleapis/python-aiplatform/pull/1196) since both are adding a new forecasting model.

It looks like in that PR the decision was made to keep `run` around to force generation of docs for this method. Along the same lines, I believe the class documentation from the `_ForecastingTrainingJob` base class' docstring is sufficient since all the changes between these two classes is the model schema being sent in the training request.

I think it's important that we maintain consistency between the `Seq2SeqPlusForecastingTrainingJob`, `AutoMLForecastingTrainingJob`, and this new TFT variant. Based on that previous PR's context, do you think it's okay to leave this as is?

Comment (sasha-gitg) [reply google/cloud/aiplatform/training_jobs.py:5386]: Sounds good. I opened to ticket to add inherited members to the docs. [b/249069357](b/249069357).

Please add a TODO comment referencing this ticket.

Comment (mikelawrence-google) [reply google/cloud/aiplatform/training_jobs.py:5210]: Thanks Sasha. I see the bug has been closed. I'll get rid of `run` for this as well.

File: tests/system/aiplatform/test_e2e_forecasting.py
Comment (mikelawrence-google) [reply tests/system/aiplatform/test_e2e_forecasting.py:46]: The backend is ready, we are just waiting for documentation and some UI pieces. It's completely usable though. My TL suggested we merge it ahead of time with the understanding that customers can use it, just without the GA guarantees at launch time (~1 month out). What do you think?

**Comment by sararob** [comment]
We are doing maintenance on our repo starting today and need to close all open PRs. When this is done next week, you'll be able to open a new PR.

**Pull request closed without merging**
